plottr/plot/mpl/autoplot.py

```diff
@@ -0,0 +1,505 @@
+    #     if len(data) > 2 and isinstance(ret, AxesImage):
+    #         if 'colorbar' not in self.plotItems[name]:
+    #             cb = self.plotItems[name]['colorbar'] = \
+    #                 self.fig.colorbar(ret, cax=ax['cax'])
+
+
+# A toolbar for setting options on the MPL autoplot
+class AutoPlotToolBar(QtWidgets.QToolBar):
+    """
+    A toolbar that allows the user to configure AutoPlot.
+
+    Currently, the user can select between the plots that are possible, given
+    the data that AutoPlot has.
+    """
+
+    #: signal emitted when the plot type has been changed
+    plotTypeSelected = Signal(PlotType)
+
+    #: signal emitted when the complex data option has been changed
+    complexPolarSelected = Signal(bool)
+
+    def __init__(self, name: str, parent: Optional[QtWidgets.QWidget] = None):
+        """Constructor for :class:`AutoPlotToolBar`"""
+
+        super().__init__(name, parent=parent)
+
+        self.plotasMultiTraces = self.addAction(get_multiTracePlotIcon(),
+                                                'Multiple traces')
+        self.plotasMultiTraces.setCheckable(True)
+        self.plotasMultiTraces.triggered.connect(
+            lambda: self.selectPlotType(PlotType.multitraces))
+
+        self.plotasSingleTraces = self.addAction(get_singleTracePlotIcon(),
+                                                 'Individual traces')
+        self.plotasSingleTraces.setCheckable(True)
+        self.plotasSingleTraces.triggered.connect(
+            lambda: self.selectPlotType(PlotType.singletraces))
+
+        self.addSeparator()
+
+        self.plotasImage = self.addAction(get_imagePlotIcon(),
+                                          'Image')
+        self.plotasImage.setCheckable(True)
+        self.plotasImage.triggered.connect(
+            lambda: self.selectPlotType(PlotType.image))
+
+        self.plotasMesh = self.addAction(get_colormeshPlotIcon(),
+                                         'Color mesh')
+        self.plotasMesh.setCheckable(True)
+        self.plotasMesh.triggered.connect(
+            lambda: self.selectPlotType(PlotType.colormesh))
+
+        self.plotasScatter2d = self.addAction(get_scatterPlot2dIcon(),
+                                              'Scatter 2D')
+        self.plotasScatter2d.setCheckable(True)
+        self.plotasScatter2d.triggered.connect(
+            lambda: self.selectPlotType(PlotType.scatter2d))
+
+        # other options
+        self.addSeparator()
+
+        self.plotComplexPolar = self.addAction('Mag/Phase')
+        self.plotComplexPolar.setCheckable(True)
+        self.plotComplexPolar.triggered.connect(self._trigger_complex_mag_phase)
+
+        self.plotTypeActions = OrderedDict({
+            PlotType.multitraces: self.plotasMultiTraces,
+            PlotType.singletraces: self.plotasSingleTraces,
+            PlotType.image: self.plotasImage,
+            PlotType.colormesh: self.plotasMesh,
+            PlotType.scatter2d: self.plotasScatter2d,
```

Comment:
exactly, shouldn't these functions be required to be implemented by a concrete AutoFigureMaker class?